Formula/top-post.rb

```diff
@@ -0,0 +1,25 @@
+require "language/node"
+
+class TopPost < Formula
+  desc "⚙️ Show Hacker News articles in the terminal"
+  homepage "https://github.com/daschaa/top-post"
+  url "https://registry.npmjs.org/top-post/-/top-post-0.0.9.tgz"
+  sha256 "cb2f030eed227c026a49864888ccec84c5ad124333294563bbe5050b1c0326cf"
+  license "MIT"
+
+  livecheck do
+    url :stable
+  end
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    output = `#{bin}/top-post --help`
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Done